src/main/groovy/org/mastercoin/test/TestSupport.groovy

```diff
@@ -265,7 +265,7 @@ trait TestSupport implements MastercoinClientDelegate {
      * @param amount     The number of units to create
      * @return The transaction hash
      */
-    Sha256Hash createProperty(Address address, Ecosystem ecosystem, PropertyType type, Long amount) {
+    Sha256Hash createProperty(Address address, Ecosystem ecosystem, PropertyType type, Number amount) {
```

Comment:
Agreed, having specific classes to represent Omni types should definitely be aimed for. I'll update or split this PR later. Regarding the constructors, with `Ecosystem` as example: I assume there can be a loss of precision due to the implicit conversion by `(short) value` or `value.shortValue()`, so I'm wondering, if instead `Number` should be used as type in the constructor which has the range checks, for example:

```java
public Ecosystem(Number value) {
    if (value < MIN_VALUE) {
        throw new NumberFormatException();
    }
    if (value > MAX_VALUE) {
        throw new NumberFormatException();
    }
    this.value = value.shortValue(); // <--- at this point doing it should be safe
}
```